tests/test_pg_exe.py

```
@@ -0,0 +1,157 @@
+
+
+@pytest.mark.skipif(platform.system() == "Windows", reason="os.access ignores X_OK on Windows")
+def test_non_executable_configured_path_is_rejected(tmp_path: Path) -> None:
+    """A configured pg_ctl that exists but cannot be run is not handed back.
+
+    It used to be accepted, deferring the failure to the version check.
+    """
+    pg_ctl = make_pg_ctl(tmp_path / "bin", mode=0o644)
+    config = make_config(str(pg_ctl))
+    with patch(f"{_PG_PROCESS}.subprocess.check_output", side_effect=FileNotFoundError("pg_config")):
+        with pytest.raises(ExecutableMissingException, match="Could not find pg_ctl"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)